### PR TITLE
chore: add map stream udf example

### PR DESCRIPTION
**Pull request by xdevxy**
Add a map stream udf example for flatmap.

### CODE REVIEW

**Comment by vigith** [comment]
shouldn't we also add a non-async example too?

File: examples/function/flatmap_stream/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Flatmap Stream Example
+
+## Build image and push
+
+```shell
+make image
+# Privilege needed.
+docker push quay.io/numaio/numaflow-python/map-flatmap-stream:latest
+```
+
```
Comment (yhl25) [inline examples/function/flatmap_stream/README.md:10]: can we use a multi-platform image build command?

**Review by yhl25 — APPROVED** [review]

File: examples/function/flatmap_stream/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[tool.poetry]
+name = "map-flatmap-stream"
+version = "0.2.4"
+description = ""
+authors = ["Numaflow developers"]
+
+[tool.poetry.dependencies]
+python = "~3.10"
+pynumaflow = "~0.4.0"
+aiorun = "^2022.11.1"
```
Comment (ab93) [inline examples/function/flatmap_stream/pyproject.toml:10]: I don't think they need to add aiorun as their dependency. Pynumaflow already has it :)

File: examples/function/flatmap_stream/example.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import aiorun
+from collections.abc import AsyncIterable
+
+from pynumaflow.function import Message, Datum, AsyncServer
+
+
+async def map_stream_handler(_: list[str], datum: Datum) -> AsyncIterable[Message]:
+    val = datum.value
```
Comment (ab93) [inline examples/function/flatmap_stream/example.py:8]: Let's add a docstring here maybe? Briefly explaining what this udf is trying to do with the streaming data

File: examples/function/flatmap_stream/README.md (hunk 1 above)
Comment (xdevxy) [reply examples/function/flatmap_stream/README.md:10]: Not sure which command exactly you are referring to? `buildx`?

**Pull request merged**